sway/tree/arrange.c

```diff
@@ -111,12 +111,19 @@ void add_gaps(struct sway_container *c) {
 		return;
 	}
 
-	if (c->type == C_WORKSPACE &&
-		!(config->edge_gaps || (config->smart_gaps && c->children->length > 1))) {
-		return;
-	}
+	double gaps = 0;
 
-	double gaps = c->has_gaps ? c->gaps_inner : config->gaps_inner;
+	if (c->type == C_WORKSPACE) {
+		if (!config->edge_gaps || (config->smart_gaps && c->children->length == 1)) {
+			return;
+		}
+		gaps = c->has_gaps ? c->gaps_outer : config->gaps_outer;
+	} else {
+		if (config->smart_gaps && container_parent(c, C_WORKSPACE)->children->length < 2) {
```

Comment:
Does `config->edge_gaps` need to be checked here?

Reply:
Edge gaps must be enabled to have outer gaps. If disabled, workspaces have no gaps because workspaces only have outer gaps.

Reply:
Ergo edge_gaps must be checked. It's cheap though.